### PR TITLE
docs: update Flyte sandbox configuration and documentation

### PR DESCRIPTION
**Pull request by jasonlai1218**
## Why are the changes needed?
Currently, [official documentation](https://docs.flyte.org/en/latest/deployment/plugins/k8s/index.html#configure-kubernetes-plugins) does not provide instructions on how to use Helm to deploy Flyte sandbox on a Kubernetes cluster and configure the Spark plugin.

## What changes were proposed in this pull request?
- Add instructions for setting up Flyte sandbox in the Kubernetes plugin documentation
- Provide example YAML files for configuring Spark in Flyte sandbox
- Update the values-override.yaml file for Flyte sandbox configuration
- Fix the incorrect command by changing `kubectl get pods -n --all-namespaces` to `kubectl get pods -n <namespace>`.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

**Comment by davidmirror-ops** [comment]
Hey Jason!

Would you be open to install flyte-binary instead?

This guide should help you gets started on an on prem K8s cluster
https://github.com/davidmirror-ops/flyte-the-hard-way/blob/main/docs/on-premises/001-configure-local-k8s.md

Then, using the Spark config for that chart should be more straightforward

That's a workaround for sure, docs for sandbox should be reproducible, I'm
working on it.

File: docs/deployment/plugins/k8s/index.rst [hunk 1]
```diff
@@ -277,6 +277,85 @@ Spin up a cluster
 
     helm repo add flyteorg https://flyteorg.github.io/flyte
 
+  .. group-tab:: Flyte sandbox
+
+    If you have installed Flyte using the `flyte-core Helm chart
```
Comment (davidmirror-ops) [inline docs/deployment/plugins/k8s/index.rst:282]: So are we using the `flyte-core` structure to configure the sandbox cluster?

File: docs/deployment/plugins/k8s/index.rst [hunk 2]
```diff
@@ -277,6 +277,85 @@ Spin up a cluster
...
+            2. ``spark_role.yaml``
+
+            .. code-block:: yaml
+
+              apiVersion: rbac.authorization.k8s.io/v1
+              kind: Role
+              metadata:
+                name: spark-role
+                namespace: "{{ namespace }}"
+              rules:
```
Comment (davidmirror-ops) [inline docs/deployment/plugins/k8s/index.rst:319]: One of the tests is complaining about a list being out of range. Not sure if it's a matter of indentation.

Also, this is the Role I've been using with flyte-core on EKS:

https://github.com/flyteorg/flyte/pull/4670/files#diff-5e363febf4c9f8a60b6f884277d82eb90f1d76e1f2a5b14037133c1c9ff36adeR574-R601

File: docs/deployment/plugins/k8s/index.rst [hunk 3]
```diff
@@ -751,6 +830,122 @@ Specify plugin configuration
...
                       container_array: k8s-array
                       spark: spark
+
+        .. group-tab:: Flyte sandbox
+
+          Create a file named ``values-override.yaml`` and add the following config to it:
+
+          .. note::
+
+            Within the flyte-binary block, the value of inline.storage.signedURL.stowConfigOverride.endpoint should be set to the corresponding node Hostname/IP on the MinIO pod if you are deploying on a Kubernetes cluster.
```
Comment (davidmirror-ops) [inline docs/deployment/plugins/k8s/index.rst:840]: Is this particular to `flyte-core`? I haven't had to use it with `flyte-binary` on-prem

Comment (jasonlai1218) [reply docs/deployment/plugins/k8s/index.rst:840]: And it is possible that the two pods are not on the same K8S node, so the settings in this [line](https://github.com/flyteorg/flyte/blob/master/charts/flyte-sandbox/values.yaml#L53) may need to be set. Node IP

Screenshot 2024-01-16 at 11 25 35 PM

Screenshot 2024-01-16 at 11 18 32 PM

Comment (jasonlai1218) [reply docs/deployment/plugins/k8s/index.rst:840]: by the way, Minio service uses NodePort
https://github.com/flyteorg/flyte/blob/master/charts/flyte-sandbox/values.yaml#L116-119

File: docs/deployment/plugins/k8s/index.rst [hunk 4]
```diff
@@ -277,6 +278,88 @@ Spin up a cluster
 
     helm repo add flyteorg https://flyteorg.github.io/flyte
 
+  .. tabs::
+
+    If you have installed Flyte using the `flyte-sandbox Helm chart<https://github.com/flyteorg/flyte/tree/master/charts/flyte-sandbox>`__, please ensure:
```
Comment (samhita-alla) [inline docs/deployment/plugins/k8s/index.rst:283]:
```suggestion
    If you have installed Flyte using the `flyte-sandbox Helm chart <https://github.com/flyteorg/flyte/tree/master/charts/flyte-sandbox>`__, please ensure:
```

**Review by davidmirror-ops — APPROVED** [review]
While I haven't tried it using this method (I mean, using the flyte-sandbox Helm chart instead of `flytectl demo start`) I see the values reflect what's expected for that chart.

**Review by Future-Outlier — APPROVED** [review]
LGTM, thanks for your effort!

**Comment by davidmirror-ops** [comment]
@jasonlai1218 Thanks so much!
BTW, consider signing up for the [community raffle](https://www.union.ai/blog-post/flyte-weekly-community-raffle-lets-soar-to-new-heights), this contribution will give you points!

**Pull request merged**